### PR TITLE
[enhancement](mysql-channel) avoid potential buffer overflow after flushing send buffer occurs IOE

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

1. Ensured always clearing send buffer after `MysqlChannel::flush` op.
2. Corrected the logic of `MysqlChannel::writeBuffer`.

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlChannel.java [hunk 1]
```diff
@@ -423,18 +426,17 @@ private void writeHeader(int length, boolean isSsl) throws IOException {
...
 
-    private void writeBuffer(ByteBuffer buffer, boolean isSsl) throws IOException {
+    private void writeBuffer(ByteBuffer buffer) throws IOException {
         if (null == sendBuffer) {
             return;
         }
-        long leftLength = sendBuffer.capacity() - sendBuffer.position();
         // If too long for buffer, send buffered data.
-        if (leftLength < buffer.remaining()) {
+        if (sendBuffer.remaining() < buffer.position()) {
```
[the remark this answers is not in the document]
Comment (TangSiyang2001) [reply fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlChannel.java]: Got confused. It should be `remaining` under read mode. Code modified.

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run feut

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**